titus_isolate/isolate/workload_manager.py

```diff
@@ -15,11 +17,19 @@ def __init__(self, cpu, cgroup_manager, allocator_class=IntegerProgramCpuAllocat
         self.__error_count = 0
         self.__added_count = 0
         self.__removed_count = 0
+        self.__allocator_call_duration_sum_secs = 0
+        self.__fallback_allocator_calls_count = 0
+        self.__time_bound_ip_allocator_solution_count = 0
 
         self.__cpu = cpu
         self.__cgroup_manager = cgroup_manager
         self.__workloads = {}
         self.__cpu_allocator = allocator_class(cpu)
+        self.__is_ip_allocator_used = False
+        self.__fallback_cpu_allocator = None
+        if isinstance(self.__cpu_allocator, IntegerProgramCpuAllocator):
+            self.__is_ip_allocator_used = True
```

Comment:
This is unfortunate, but not a blocker for the PR.  I'll think about this for a bit.  It would be better if we didn't have to treat allocators specially.